vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java

```diff
@@ -421,4 +423,14 @@ private void doDisableOnClick() {
                     }));
         }
     }
+
+    @Override
+    protected void onAttach(AttachEvent attachEvent) {
+        initDisableOnClick();
+    }
+
+    @Override
+    protected void onDetach(DetachEvent detachEvent) {
+        disableOnClickConfigured = false;
```

Comment:
I think the `disableOnClickConfigured` property can be removed. It seems the whole purpose of the flag was to check that the listener is not configured multiple times, when calling `setDisabledOnClick` multiple times in a row. Now that we always execute `initDisableOnClick` on attach, it will always be true after attaching the component, plus there is no need to run `initDisableOnClick` again in `setDisabledOnClick`.

So in summary we can:
- remove `disableOnClickConfigured`
- remove the `initDisableOnClick` call here: https://github.com/vaadin/flow-components/blob/master/vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java#L323
- remove the onDetach listener

Reply:
Didn't realized ```disabledOnClickConfigured``` flag was not necessary. I updated the branch with the proposed changes.